src/test/java/uk/nhs/tis/sync/job/PostFundingSyncJobTest.java

```diff
@@ -0,0 +1,113 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+class PostFundingSyncJobTest {
+
+  @Mock
+  private EntityManager entityManager;
+
+  @Mock
+  private ObjectMapper objectMapper;
+
+  private PostFundingSyncJob postFundingSyncJob;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    postFundingSyncJob = new PostFundingSyncJob(objectMapper);
+  }
+
+  @Test
+  void testBuildQueryForDate() {
+    LocalDate dateOfChange = LocalDate.now();
+    String expectedQuery =
+        "SELECT DISTINCT p.id FROM Post p JOIN ( SELECT postId FROM PostFunding WHERE startDate IS NOT NULL AND (endDate = '"
+            + dateOfChange.minusDays(1).format(DateTimeFormatter.ISO_LOCAL_DATE)
+            + "' OR endDate IS NULL) GROUP BY postId HAVING COUNT(id) > 0) pf ON p.id = pf.postId ORDER BY p.id LIMIT "
+            + PostFundingSyncJob.DEFAULT_PAGE_SIZE;
+    String actualQuery = postFundingSyncJob.buildQueryForDate(dateOfChange);
+    assertEquals(expectedQuery, actualQuery);
+  }
+
+  @Test
+  void testConvertData() {
+    Post post = new Post();
+    post.setId(1L);
+    PostFunding postFunding = new PostFunding();
+    postFunding.setId(999L);
+    post.fundingStatus(Status.CURRENT);
+    Set<PostFunding> postFundingSet = new HashSet<>(Arrays.asList(postFunding));
+    post.setFundings(postFundingSet);
+
+    when(entityManager.find(eq(Post.class), anyLong())).thenReturn(post);
+
+    Set<Post> entitiesToSave = new HashSet<>();
+    List<Long> entityData = Arrays.asList(1L);
+    int result = postFundingSyncJob.convertData(entitiesToSave, entityData, entityManager);
+
+    assertEquals(0, result);
+    assertTrue(entitiesToSave.contains(post));
+    assertEquals(Status.INACTIVE, post.getFundingStatus());
+  }
+
+  @Test
+  void testPostFundingWillRemainCurrentWithMulitiplePostFunding() {
+    Post post = new Post();
+    post.setId(1L);
+    PostFunding postFunding1 = new PostFunding();
+    postFunding1.setId(999L);
+    PostFunding postFunding2 = new PostFunding();
+    postFunding2.setId(1000L);
+    post.fundingStatus(Status.CURRENT);
+    Set<PostFunding> postFundingSet = new HashSet<>(Arrays.asList(postFunding1, postFunding2));
+    post.setFundings(postFundingSet);
+
+    when(entityManager.find(eq(Post.class), anyLong())).thenReturn(post);
+
+    Set<Post> entitiesToSave = new HashSet<>();
+    List<Long> entityData = Arrays.asList(1L);
+    int result = postFundingSyncJob.convertData(entitiesToSave, entityData, entityManager);
+
+    assertEquals(0, result);
+    assertTrue(entitiesToSave.contains(post));
+    assertEquals(Status.CURRENT, post.getFundingStatus());
+  }
+
+  @Test
+  void testHandleData() {
+    Post post = new Post();
+    Set<Post> dataToSave = new HashSet<>();
+    dataToSave.add(post);
```

Comment:
There are good initializers in `Collection` and `Set` that allow you to do this on 1 line.

Reply:
I think these lines are not there in the final version of the PR